### PR TITLE
[LabKey UX] Edit standard settings name/label/type - Domain API updates

### PR DESCRIPTION
**Pull request by labkey-martyp**
Add domainId to Domain.get and Domain.save.  Needed for new domain designer UI.

### CODE REVIEW

File: api/gwtsrc/org/labkey/api/gwt/client/assay/model/GWTPropertyDescriptorMixin.java [hunk 1]
```diff
@@ -16,7 +16,9 @@
         "setDimension",
         "setExcludeFromShifting",
         "lookupDescription",
-        "fileType"
+        "fileType",
+        "updatedField",
+        "newField"
```
Comment (cnathe) [inline api/gwtsrc/org/labkey/api/gwt/client/assay/model/GWTPropertyDescriptorMixin.java]: @labkey-martyp What is this for? It seems unrelated to this commit / set of changes?

**Comment by cnathe** [comment]
@labkey-martyp Looks like there are some BVT failures for this branch on TeamCity that would be good to triage to see if they are related:
https://teamcity.labkey.org/viewLog.html?buildId=728107&buildTypeId=bt7&tab=buildResultsDiv
https://teamcity.labkey.org/viewLog.html?buildId=728105&tab=buildResultsDiv&buildTypeId=bt8

File: api/gwtsrc/org/labkey/api/gwt/client/assay/model/GWTPropertyDescriptorMixin.java (hunk 1 above)
Comment (labkey-martyp) [reply api/gwtsrc/org/labkey/api/gwt/client/assay/model/GWTPropertyDescriptorMixin.java]: As discussed these fields need to be identified for Jackson marshaller to ignore them.
